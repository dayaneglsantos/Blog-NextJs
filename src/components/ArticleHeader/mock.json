{
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "title": "TÃ­tulo da postagem",
  "excerpt": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero ratione quisquam alias doloribus veniam dolore. Ea temporibus illum numquam ab quo.",
  "image": "/image.com",
  "author": {
    "slug": "dayane-santos",
    "displayName": "Dayane Santos"
  },
  "categories": [
    {
      "slug": "react",
      "displayName": "React"
    },
    {
      "slug": "js",
      "displayName": "Javascript"
    }
  ]
}
