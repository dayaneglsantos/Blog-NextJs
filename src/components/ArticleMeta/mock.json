{
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "darkTheme": false,
  "author": {
    "slug": "dayane-santos",
    "displayName": "Dayane Santos"
  },
  "categories": [
    {
      "slug": "react",
      "displayName": "React"
    },
    {
      "slug": "js",
      "displayName": "Javascript"
    }
  ]
}
