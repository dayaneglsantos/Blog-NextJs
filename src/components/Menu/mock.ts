import { MenuType } from '.'

export default {
  links: [
    {
      link: '/',
      text: 'Home',
      newTab: true
    },
    {
      link: '/about',
      text: 'About',
      newTab: true
    },
    {
      link: '/posts',
      text: 'Posts',
      newTab: true
    },
    {
      link: '/fallow-us',
      text: 'Fallow us',
      newTab: true
    },
    {
      link: '/contact',
      text: 'Contact',
      newTab: true
    }
  ]
} as MenuType
