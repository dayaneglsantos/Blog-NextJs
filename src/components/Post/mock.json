{
  "title": "Post 2",
  "slug": "post-2",
  "alternativeText": "Foto 2",
  "image": "https://res.cloudinary.com/dbmuc0tzr/image/upload/v1703092587/img5_565f978262.jpg",
  "excerpt": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero ratione quisquam alias doloribus veniam dolore",
  "textContent": "<p>Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, commodo vitae, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. Donec non enim in turpis pulvinar facilisis. Ut felis. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus</p><ul><li>Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.</li><li>Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus.</li><li>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.</li><li>Pellentesque fermentum dolor. Aliquam quam lectus, facilisis auctor, ultrices ut, elementum vulputate, nunc.</li></ul>",
  "allowComments": true,
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "cover": "/cover.com.br/img",
  "altText": "Alt text do cover",
  "id": "1",

  "categories": [
    {
      "displayName": "React",
      "slug": "react"
    },
    {
      "displayName": "Front-end",
      "slug": "fron-end"
    }
  ],
  "tags": [
    {
      "displayName": "Tag 1",
      "slug": "tag-1"
    },
    {
      "displayName": "Tag 2",
      "slug": "tag-2"
    }
  ],
  "author": {
    "displayName": "Dayane Santos",
    "slug": "dayane-santos"
  }
}
