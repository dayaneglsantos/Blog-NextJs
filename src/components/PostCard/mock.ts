import { data } from '../../Api/data.json'

export default data.posts.data[0].attributes
