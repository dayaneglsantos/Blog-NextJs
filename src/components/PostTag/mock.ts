import data from '../../Api/data.json'

export default data.data.posts.data[0].attributes.tags
