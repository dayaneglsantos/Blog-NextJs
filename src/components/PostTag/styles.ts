'use client'
import styled from 'styled-components'

export const PostTagContainer = styled.div`
  margin: 16px auto;
  max-width: 600px;
  width: 100%;
  color: ${({ theme }) => theme.colors.secondaryFont};

  a::after {
    content: ', ';
  }
  a::before {
    content: ' ';
  }
  a:last-child::after {
    content: '';
  }
`
