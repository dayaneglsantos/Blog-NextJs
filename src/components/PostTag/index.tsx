import Link from 'next/link'
import { Tag } from '../../shared-types/tags'
import { PostTagContainer } from './styles'

export type TagType = {
  tags?: Tag[]
}

const PostTag = ({ tags = [] }: TagType) => {
  if (!tags.length) {
    return null
  }

  return (
    <>
      <PostTagContainer>
        Tags:
        {tags.map((tag) => (
          <Link key={tag.slug} href={`/tag/${tag.slug}`}>
            {tag.displayName}
          </Link>
        ))}
      </PostTagContainer>
    </>
  )
}

export default PostTag
