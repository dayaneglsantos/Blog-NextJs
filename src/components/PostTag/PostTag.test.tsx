import { screen } from '@testing-library/react'
import PostTag from '.'
import mock from './mock'
import { renderTheme } from '../../Utils/render-theme'

describe('<PostTag />', () => {
  it('should render the correct number of links', () => {
    renderTheme(<PostTag tags={mock} />)

    const link = screen.getAllByRole('link')
    expect(link).toHaveLength(2)

    const linkJest = screen.getByRole('link', { name: 'Jest' })
    expect(linkJest).toBeInTheDocument()
  })
  it('should render with no tags', () => {
    renderTheme(<PostTag />)

    const link = screen.queryByRole('link')
    expect(link).not.toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostTag tags={mock} />)

    expect(container).toMatchSnapshot()
  })
})
