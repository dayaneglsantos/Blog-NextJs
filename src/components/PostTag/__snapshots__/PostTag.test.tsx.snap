// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTag /> should match snapshot 1`] = `
.c0 {
  margin: 16px auto;
  max-width: 600px;
  width: 100%;
  color: #495E57;
}

.c0 a::after {
  content: ', ';
}

.c0 a::before {
  content: ' ';
}

.c0 a:last-child::after {
  content: '';
}

<div>
  <div
    class="c0"
  >
    Tags:
    <a
      href="/tag/jest"
    >
      Jest
    </a>
    <a
      href="/tag/next-js"
    >
      Next
    </a>
  </div>
</div>
`;
